### PR TITLE
Recurring Billing Agreement Add total amount.

### PR DESCRIPTION
**Pull request by warmkesselj**
### Summary of changes

Additional field needed to be added to the iOS feature branch, total_amount. 

### Authors
@jwarmkessel 
@KunJeongPark

### CODE REVIEW

**Review by sshropshire — APPROVED** [review]

File: Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift [hunk 1]
```diff
@@ -37,7 +39,8 @@ public struct BTPayPalRecurringBillingDetails {
         oneTimeFeeAmount: String?,
         shippingAmount: String?,
         productAmount: String?,
-        taxAmount: String?
+        taxAmount: String?,
+        totalAmount: String?
```
Comment (sarahkoop) [inline Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift]: Is this based on the final GW PR for this? I think the `productAmount` field also changed and there is a `productPrice` filed now

**Review by scannillo — APPROVED** [review]

File: Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift (hunk 1 above)
Comment (warmkesselj) [reply Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift]: Good catch. The param being sent is `"product_price"` while the interface is `productAmount`. I'll update this to use `productPrice`.
```
if let productAmount {
            parameters["product_price"] = productAmount
        }
```

Comment (scannillo) [reply Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift]: The SDK is sending `product_price` in the JSON blob to the GW.

I intentionally opted to name the public SDK property `amount` instead since it is more consistent with how we name dollar amounts throughout the SDK. We don't use `price` anywhere else.

File: Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift [hunk 2]
```diff
@@ -12,8 +12,9 @@ public struct BTPayPalRecurringBillingDetails {
     private let productQuantity: Int?
     private let oneTimeFeeAmount: String?
     private let shippingAmount: String?
-    private let productAmount: String?
+    private let productPrice: String?
```
Comment (jaxdesmarais) [inline Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift:15]: Sounds like we want to revert this change and keep `productAmount` based on this comment: https://github.com/braintree/braintree_ios/pull/1373#discussion_r1688004735

File: Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift [hunk 3]
```diff
@@ -36,8 +38,9 @@ public struct BTPayPalRecurringBillingDetails {
         productQuantity: Int?,
         oneTimeFeeAmount: String?,
         shippingAmount: String?,
-        productAmount: String?,
-        taxAmount: String?
+        productPrice: String?,
+        taxAmount: String?,
+        totalAmount: String
```
Comment (jaxdesmarais) [inline Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift:43]: Total amount is required here and optional above. Should these be the same?

File: Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift (hunk 1 above)
Comment (sarahkoop) [reply Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift]: Ahh sorry my mistake!

File: Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift (hunk 3 above)
Comment (warmkesselj) [reply Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift:43]: Yes, it should be a required property.

File: Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift [hunk 4]
```diff
@@ -14,6 +14,7 @@ public struct BTPayPalRecurringBillingDetails {
     private let shippingAmount: String?
     private let productAmount: String?
     private let taxAmount: String?
+    private let totalAmount: String?
```
Comment (jaxdesmarais) [inline Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift:17]:
```suggestion
    private let totalAmount: String
```

Comment (jaxdesmarais) [reply Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift:17]: Lets also move this to the top with the other required params

File: Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift (hunk 3 above)
Comment (jaxdesmarais) [reply Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift:43]: Cool, can we move it to the top of the init with the other required params? Along with moving up the docstring.

Comment (warmkesselj) [reply Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift]: ✅

File: Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift [hunk 5]
```diff
@@ -86,6 +90,8 @@ public struct BTPayPalRecurringBillingDetails {
             parameters["tax_amount"] = taxAmount
         }
         
+        parameters["total_amount"] = totalAmount
```
Comment (jaxdesmarais) [inline Sources/BraintreePayPal/RecurringBillingMetadata/BTPayPalRecurringBillingDetails.swift:93]: nit: we should be able to add this directly to the base parameters in line 62

**Review by agedd — APPROVED** [review]

**Review by jaxdesmarais — APPROVED** [review]
